### PR TITLE
Classifier modules

### PR DESCRIPTION
**Pull request by zbalkhy**
Summary of changes:

1. inits with datasets for each label, filters, and features. The class has getter/setter/etc functions for each. 
2. There is a training pipeline, which gets each dataset, applies filters, extracts features, and attaches the corresponding label. 
3. These new features are used to train a sklearn LogisticRegression model. 
4. Right now, I only have it support 1D. 
5. This training process is not used anywhere in the interface at the moment.
6. The interface (shown in screenshot)  allows selection of multiple datasets for label0 and label1, features (userModel now contains the default features at initialization), and filters. 
7. Clicking the 'create classifier' button adds a classifier with the given name and displays it on the right side, but these classifiers do not have any intractability at the moment. 

image

### CODE REVIEW

File: src/classifier.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    # -----------------------
+    def generate_features(self, data: np.ndarray) -> np.ndarray:
+        # apply filters
+        for flt in self.filters:
+            data = flt.apply(data)
+
+        # extract features per channel
+        feature_vectors = []
+        for feature in self.features:
+            # TODO: instead of hard-coding SAMPLING_FREQ, have the Features class
```
Comment (zbalkhy) [inline src/classifier.py:77]: Agreed. Whether its a wrapper class or something stored implicitly in the dataset via timestamps, we will need a way to get fs for each dataset individually

File: src/classifier.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        )
+        self.model.fit(X, y)
+
+    # -----------------------
+    # Prediction
+    # -----------------------
+    def predict_sample(self, sample: np.ndarray, fs: int) -> int:
+        if self.model is None:
+            raise RuntimeError("Model not trained yet.")
+        features = self.extract_features(sample, fs).reshape(1, -1)
```
Comment (zbalkhy) [inline src/classifier.py:114]: Where is this function defined?

File: src/main.py [hunk 3]
```diff
@@ -39,6 +41,18 @@ def on_closing():
     user_model = UserModel()
     data_stream = SoftwareStream("software_stream_test", StreamType.SOFTWARE)
     user_model.add_stream(data_stream)
+    
+    default_features = [
```
Comment (zbalkhy) [inline src/main.py:45]: nice, thanks for adding this
